### DISCUSSION TITLE
saspy sasstat() results as text?

### DISCUSSION THREAD

**Discussion by mrmorgan17** [sassoftware/saspy#484] [Q&A] [answered]
I want the results of my sas session to be in text format so I initialized my session as follows:
```
import saspy
sas = saspy.SASsession(results = 'TEXT')
sas
```

Then, using saspy, I fit a mixed model using the following code:
```
stat = sas.sasstat()
stat_results = stat.mixed(data = 'experiment', cls = 'A B block', model = 'y = A*B / noint s ddfm=satterth', random = 'block')
```

`dir(stat_results)` returns the following:
```
['BOXPLOT', 'CLASSLEVELS', 'CONVERGENCESTATUS', 'COVPARMS', 'DIMENSIONS', 'FITSTATISTICS', 'ITERHISTORY', 'LOG', 'MODELINFO', 'NOBS', 'PEARSONPANEL', 'RESIDUALPANEL', 'SOLUTIONF', 'STUDENTPANEL', 'TESTS3']
```

Which suggests that the model was fit

But then when I run `stat_results.FITSTATISTICS` I get this output:
```
<IPython.core.display.HTML object>
``` 

It seems that my SASsession is reverting to results = 'Jupyter' somehow. Would anyone know what is going on?

I am using reticulate to use saspy in RStudio but just wanted to know if there was still a way to view the objects of `stat_results` in text format.

Additionally, I am wondering if there is a way to save `stat_results.FITSTATISTICS` as a Pandas data frame

Thanks!

**Comment by tomweber-sas** [comment, marked as answer]
Hey, you're right, any tabular output from the analytic methods isn't being rendered as TEXT when results='text' (non-tabular, like plots graphs ..., will only be returned as HTML). But, they will be returned as Pandas DataFrames when you use the default results='PANDAS'. 

```
>>> res.FITSTATISTICS
<IPython.core.display.HTML object>
>>> sas.set_results('pandas')
>>> res.FITSTATISTICS
           Label1    cValue1     nValue1    Label2 cValue2   nValue2
0        Root MSE   41.10897   41.108975  R-Square  0.6755  0.675460
1  Dependent Mean  215.87793  215.877934  Adj R-Sq  0.6739  0.673926
2       Coeff Var   19.04269   19.042694       NaN     NaN  0.000000
>>> df = res.FITSTATISTICS
>>> type(df)
<class 'pandas.core.frame.DataFrame'>
>>>
```

**Reply by mrmorgan17** [reply]
Worked like a charm, thanks!
